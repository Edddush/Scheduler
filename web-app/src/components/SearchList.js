import React, { useContext } from "react";
import CourseTable from "./CourseTable";
import { ScheduleContext } from '../context/ScheduleContext';
import { addCourse } from "./ScheduleFunctions";
import Pagination from "./Pagination";
import CircularProgress from '@mui/material/CircularProgress';
import Box from '@mui/material/Box';


function SearchList({
  currentPage,
  setCurrentPage,
  searchResults,
  setSearchResults,
  searchInput,
  loading
}) {
  const columns = [
    { header: "Course Name", accessProperty: "course_name" },
    { header: "Subject", accessProperty: "subject" },
    { header: "Course Code", accessProperty: "course_code" },
    { header: "Section Number", accessProperty: "section_number" },
    { header: "Term", accessProperty: "term" },
    { header: "Meeting Information", accessProperty: "meeting_info_string" },
  ];

  const { setSchedules } = useContext(ScheduleContext);
  const limit = 10;

  const addCourseCallback = (row) => {
    addCourse(setSchedules, row);
    setSearchResults((prev) => prev.filter((value) => value.uid !== row.uid));
  };

  const callbackCol = {
    header: "Add Section",
    text: "Add Section",
    callback: addCourseCallback,
  };


  const lastIndex = currentPage * limit;
  const firstIndex = lastIndex - limit;
  const resultsPerPage = searchResults.slice(firstIndex, lastIndex);


  return (
    <div>
      <h3 id="course_header">Search Results</h3>
      <h6 id="course_header_term">Note: Search Results are for {searchInput.term}. To change terms, please Add a New Schedule below using the + button for that term.</h6>
      {
        loading ? (
          <>
            <h4>Loading...</h4>
            <Box sx={{ display: 'flex' }}>
              <CircularProgress />
            </Box>
          </>
        ) : null
      }
      {
        loading === false ? (
          <>
            <CourseTable
              columns={columns}
              callbackCol={callbackCol}
              data={resultsPerPage}
            />
            <Pagination
              currentPage={currentPage}
              max={searchResults.length}
              limit={limit}
              onPageClick={(page) => setCurrentPage(page)}
            />
          </>

        ) : null
      }
    </div>
  );
}

export default SearchList;
