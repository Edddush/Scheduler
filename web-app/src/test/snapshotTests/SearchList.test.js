import React from 'react';
import SearchList from '../../components/SearchList';
import renderer from 'react-test-renderer';
import { fullCourseSet, partialCourseSet, conflictingCourseSet } from './test.data'
import ScheduleProvider from '../../context/ScheduleContext'

it('renders with full courses', () => {
  var obj = {term:"Fall 2022"}
  const tree = renderer
    .create(
      <ScheduleProvider>
        <SearchList searchResults={fullCourseSet} searchInput={obj} />
      </ScheduleProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with partial courses', () => {
  var obj = {term:"Fall 2022"}
  const tree = renderer
    .create(
      <ScheduleProvider>
        <SearchList searchResults={partialCourseSet} searchInput={obj}/>
      </ScheduleProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with full conflicting courses', () => {
  var obj = {term:"Fall 2022"}
  const tree = renderer
    .create(
      <ScheduleProvider>
        <SearchList searchResults={conflictingCourseSet} searchInput={obj}/>
      </ScheduleProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});