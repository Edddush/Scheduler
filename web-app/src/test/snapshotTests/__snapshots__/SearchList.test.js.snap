// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with full conflicting courses 1`] = `
<div>
  <h3
    id="course_header"
  >
    Search Results
  </h3>
  <h6
    id="course_header_term"
  >
    Note: Search Results are for 
    Fall 2022
    . To change terms, please Add a New Schedule below using the + button for that term.
  </h6>
</div>
`;

exports[`renders with full courses 1`] = `
<div>
  <h3
    id="course_header"
  >
    Search Results
  </h3>
  <h6
    id="course_header_term"
  >
    Note: Search Results are for 
    Fall 2022
    . To change terms, please Add a New Schedule below using the + button for that term.
  </h6>
</div>
`;

exports[`renders with partial courses 1`] = `
<div>
  <h3
    id="course_header"
  >
    Search Results
  </h3>
  <h6
    id="course_header_term"
  >
    Note: Search Results are for 
    Fall 2022
    . To change terms, please Add a New Schedule below using the + button for that term.
  </h6>
</div>
`;
